src/Http/Controllers/Web/PostController.php

```diff
@@ -21,10 +21,10 @@ class PostController extends BaseController
 {
     public function show(Request $request, Thread $thread, string $postSlug, Post $post): View
     {
-        event(new UserViewingPost($request->user(), $post));
-
-        $thread = $post->thread;
         $category = $thread->category;
+        if ($category->is_private && (! $request->user() || ! $request->user()->can('view', $category))) abort(404);
```

Comment:
This is a smidgen more efficient, not that it particularly matters...
```
if ($category->is_private && !($request->user() && $request->user()->can('view', $category)))
    abort(404);
```
Keep what's clearer for you logic-wise.

I would also be inclined to have the conditional action on a separate line as it feels clearer that it's an early exit, for all intents and purposes.